vowpalwabbit/learner.h

```diff
@@ -115,6 +115,17 @@ struct save_metric_data
   fn save_metric_f = nullptr;
 };
 
+// used for automl
+struct multiconfig_data
```

Comment:
What is a multiconfig? Can you please document this function type here and/or [here](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/What-is-a-learner%3F)

Reply:
it's a proposal that goes hand in hand with automl reduction but there isn't a concrete implementation of this functionality right now. The goal is to have a way to signal the stack to copy 'state' specific to offsets that the reduction has, permitting for a single stack to keep track of multiple 'states'. In this case an upper reduction can signal the lower stack to clear offset Foo or copy the values from offset Bar. These offsets should match the offsets the reduction is receiving when being called on learn/predict, which would be the way for this theoretical reduction to keep some state per different offset received (if needed).

Reply:
Is the registering of multiple states done at setup time or is it dynamic?

Reply:
it should be done at setup, yet its something that right now the lower reductions don't have any knowledge of. We could figure out a way to communicate this, but we don't have a lower reduction right now that is aware of this 'offset' specific state. This would be the next steps of this PR, I assume.